test/interval_tests/consistency.jl

```diff
@@ -388,4 +388,20 @@ setprecision(Interval, Float64)
 
     end
 
+    @testset "Type stability" begin
```

Comment:
Some observations on the tests:
- Line: 398: `@inferred op(a, b)`; I think it should read `@inferred op(x, y)`.
- I think there should be tests involving `/` with intervals having `zero(T)` in one of the edges, since this PR changes precisely that function. The same holds for `inv` and `extended_div`.
- Similarly, I think there should be some tests for `mid`, which is also affected by this PR